### PR TITLE
feat(forms): add router support

### PR DESCRIPTION
**Pull request by tujoworker**
This PR adds a couple of hooks for easy router integration. It also provides a solution to be used without a router.

Only the used code will be included in the application bundle. 

The hooks docs can be [found here](https://eufemia-git-feat-steps-router-support-eufemia.vercel.app/uilib/extensions/forms/extended-features/#uilibextensionsformsextended-featuresstepslocation-hooks).

### CODE REVIEW

File: packages/dnb-eufemia/src/extensions/forms/Steps/hooks/useStep.tsx [hunk 1]
```diff
@@ -1,15 +1,34 @@
...
     useRef<ReturnType<typeof useSharedState<StepsContextState>>>(null)
   sharedDataRef.current = useSharedState<StepsContextState>(
     id ? id + '-steps' : undefined
   )
 
-  const context = useContext(StepsContext)
+  useEffect(() => {
+    sharedDataRef.current.extend({
+      onStepChange,
+    } as unknown as StepsContextState) // Internal type
```
Comment (AadneRo) [inline packages/dnb-eufemia/src/extensions/forms/Steps/hooks/useStep.tsx:30]: Why do we need to type this twice?

**Review by AadneRo — APPROVED** [review]

File: packages/dnb-eufemia/src/extensions/forms/Steps/hooks/useStep.tsx (hunk 1 above)
Comment (tujoworker) [reply packages/dnb-eufemia/src/extensions/forms/Steps/hooks/useStep.tsx:30]: Yes 👍

**Comment by tujoworker** [comment]
I tested it with Next.js as well. It works as expected.

**Pull request merged**